tests/main/netplan-apply/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Ensure that netplan apply works with network-setup-control
+
+details: |
+    Netplan apply is used to apply network configuration to the system
+
+# run on all classic ubuntu LTS and current dev systems 16.04+
+systems: 
+    - ubuntu-16*
+    - ubuntu-18*
+    - ubuntu-19*
+    - ubuntu-20*
+
+debug: |
+
+
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    snap install test-snapd-netplan-apply --edge
+
+    echo "temporarily installing netplan D-Bus service until SRU is done..."
+    # for now, until netplan's D-Bus has been SRU'd to the netplan.io debian 
+    # package, we will just manually install the netplan D-Bus service units
+    # into the machine from the snap and install netplan.io from the archive
+    apt update
+
+    # the netplan package was renamed nplan -> netplan.io in bionic
+    case "$SPREAD_SYSTEM" in
+        ubuntu-16*)
+            NETPLAN_APT_PKG=nplan;;
+        *)
+            NETPLAN_APT_PKG=netplan.io;;
+    esac
+
+    apt install -y $NETPLAN_APT_PKG
```

Comment:
I think the typical way to test this is to write a little dbus service in python and run it for the test via systemd-run. Take a look at tests/main/document-portal-activation for examples.

Then `test-snapd-netplan-apply` could a bit of Python code that calls `io.netplan.Netplan.Apply()`, which fails predictably until we connect the interface.

Reply:
I was trying to avoid writing artificial tests which just test the policy, because it's important here not that specifically this D-Bus rule works, but that the total operation of `netplan apply` actually works. While your suggestion would probably be a good enough test, there's possibilities that it could be a bug in the test showing a false positive and the real netplan doesn't work but our artificial test does. It also would inform us more quickly of changes/bugs in upstream netplan that would break snap applications.

Reply:
I agree that it probably is ok to use the real netplan on the caller side. However, I think that the daemon side, should rather be a mocked service in case netplan (the daemon) does something awkward and we lose connectivity with the node.

Reply:
Okay, that makes more sense. I will adjust it.

Reply:
Okay, this has been implemented